java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/map/TieredStoreMapTest.java

```diff
@@ -0,0 +1,191 @@
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateByteArray;
+
+/**
+ * This test is running as part of release verification simulator test. Hence every change in this class should be
+ * discussed with QE team since it can affect release verification tests.
+ */
+public class TieredStoreMapTest extends HazelcastTest {
+    // properties
+    public int keyDomain = 1_500_000;
+    public int minValueByteArrayLength = 1;
+    public int maxValueByteArrayLength = 150_000;
+    public boolean clearTsDirectoryOnPrepare = true;
+    public boolean fillOnPrepare = true;
+    public boolean destroyOnExit = true;
+
+    public KeyLocality keyLocality = SHARED;
+
+    private IMap<Integer, byte[]> map;
+    private int[] keys;
+    private IExecutorService executor;
+
+    @Setup
+    public void setUp() throws ReflectiveOperationException {
+        keys = generateIntKeys(keyDomain, keyLocality, targetInstance);
+        executor = targetInstance.getExecutorService(name);
+        map = targetInstance.getMap(name);
+        Assert.assertTrue("Disk Tier Config must be enabled for map: " + name, isTsEnabledForMap());
+    }
+
+    @Prepare(global = true)
+    public void prepare() {
+        if (clearTsDirectoryOnPrepare) {
+            clearTsDirectory();
+        }
+        if (fillOnPrepare) {
+            fillMap();
+        }
+    }
+
+    @Verify
+    public void verify() {
+        Capacity hybridLogLength = Capacity.of(getHybridLogLengthSum(), BYTES);
+        logger.info("Sum of hybrid log length from all members: " + hybridLogLength.toPrettyString());
+        assertTrue("Hybrid log is equal to 0. Tiered Storage was not used.", hybridLogLength.bytes() > 0);
+    }
+
+    @Teardown(global = true)
+    public void tearDown() {
+        if (destroyOnExit) {
+            map.destroy();
+        }
+        executor.shutdown();
+    }
+
+    @TimeStep(prob = -1)
+    public void get(ThreadState state) {
+        map.get(state.randomKey());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void put(ThreadState state) {
+        map.put(state.randomKey(), state.randomValue());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void putIfAbsent(ThreadState state) {
+        map.putIfAbsent(state.randomKey(), state.randomValue());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void set(ThreadState state) {
+        map.set(state.randomKey(), state.randomValue());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void remove(ThreadState state) {
+        map.remove(state.randomKey());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void delete(ThreadState state) {
+        map.delete(state.randomKey());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void setThenDelete(ThreadState state) {
+        map.set(state.randomKey(), state.randomValue());
+        map.delete(state.randomKey());
+    }
+
+    private boolean isTsEnabledForMap() {
+        return getMapConfig().getTieredStoreConfig().getDiskTierConfig().isEnabled();
```

Comment:
I think we should use `getMapConfig().getTieredStoreConfig().isEnabled()` instead. The above check would pass even if the map is not configured for TS.

Reply:
Right, changed. Thanks.